### DISCUSSION TITLE
Private Instance with Let's Encrypt Certs (Proxmox LXC) - Issues with accessing via HTTPS

### DISCUSSION THREAD

**Discussion by colin19889** [dani-garcia/vaultwarden#4608] [Q&A]
I'm in the process of moving my Vaultwarden instance from my Synology NAS to my Proxmox Server. I have Vaultwarden installed on a Linux LXC (using [tteck's script](https://helper-scripts.com/scripts?id=Vaultwarden))and have migrated my existing database. I can access the web UI from the local IP but want to use my existing Cloudflare domain to access it via HTTPS (local only). When it was running on my Synology I used DuckDNS and Let's Encrypt but now want to use Cloudflare. I've followed [this guide](https://github.com/dani-garcia/vaultwarden/wiki/Running-a-private-vaultwarden-instance-with-Let's-Encrypt-certs) with Caddy installed on the same LXC. I'm getting errors, and not being familiar with Caddy, I'm not entirely sure what is going on. My Caddyfile, caddy.env and output logs from Caddy are as follows. Is someone able to point me in the right direction? (Domain, cert_hash, unique_id etc.) have been removed.

My Caddyfile contains

`{$DOMAIN}:443 {`
   ` tls {`
        `dns cloudflare {$CLOUDFLARE_API_TOKEN}`
   ` }`
   ` reverse_proxy localhost:8080`
`}`

My caddy.env file contains

`DOMAIN=vw.mydomain.xyz`
`CLOUDFLARE_API_TOKEN=my_api_token`

The output logs from Caddy are as follows,

`root@vaultwarden:~/.local/share/caddy# caddy run --envfile /root/.local/share/caddy/caddy.env`
`2024/06/03 17:32:56.085 INFO    using adjacent Caddyfile`
`2024/06/03 17:32:56.086 INFO    adapted config to JSON  {"adapter": "caddyfile"}`
`2024/06/03 17:32:56.086 WARN    Caddyfile input is not formatted; run 'caddy fmt --overwrite' to fix inconsistencies    {"adapter": "caddyfile", "file": "Caddyfile", "line": 2}`
`2024/06/03 17:32:56.089 INFO    admin   admin endpoint started  {"address": "localhost:2019", "enforce_origin": false, "origins": ["//localhost:2019", "//[::1]:2019", "//127.0.0.1:2019"]}`
`2024/06/03 17:32:56.089 INFO    http.auto_https server is listening only on the HTTPS port but has no TLS connection policies; adding one to enable TLS {"server_name": "srv0", "https_port": 443}`
`2024/06/03 17:32:56.089 INFO    http.auto_https enabling automatic HTTP->HTTPS redirects        {"server_name": "srv0"}`
`2024/06/03 17:32:56.089 INFO    http    enabling HTTP/3 listener        {"addr": ":443"}`
`2024/06/03 17:32:56.089 INFO    failed to sufficiently increase receive buffer size (was: 208 kiB, wanted: 7168 kiB, got: 416 kiB). See https://github.com/quic-go/quic-go/wiki/UDP-Buffer-Sizes for details.`
`2024/06/03 17:32:56.089 INFO    http.log        server running  {"name": "srv0", "protocols": ["h1", "h2", "h3"]}`
`2024/06/03 17:32:56.089 INFO    http.log        server running  {"name": "remaining_auto_https_redirects", "protocols": ["h1", "h2", "h3"]}`
`2024/06/03 17:32:56.089 INFO    http    enabling automatic TLS certificate management   {"domains": ["vw.mydomain.xyz"]}`
`2024/06/03 17:32:56.090 INFO    autosaved config (load with --resume flag)      {"file": "/root/.config/caddy/autosave.json"}`
`2024/06/03 17:32:56.090 INFO    serving initial configuration`
`2024/06/03 17:32:56.090 WARN    tls     ARI window or selected renewal time changed     {"identifiers": ["vw.mydomain.xyz"], "cert_hash": "HASH_HERE", "ari_unique_id": "UNIQUE_ID_HERE", "cert_expiry": "2024/08/29 18:22:51.000", "prev_start": "2024/07/29 18:42:51.000", "next_start": "0001/01/01 00:00:00.000", "prev_end": "2024/07/31 18:42:51.000", "next_end": "0001/01/01 00:00:00.000", "prev_selected_time": "2024/07/31 17:27:59.000", "next_selected_time": "0001/01/01 00:00:00.000", "explanation_url": ""}`
`2024/06/03 17:32:56.090 ERROR   tls     updating ARI upon managing      {"error": "could not fully update ACME renewal info: either no ACME issuer configured for certificate, or all failed (make sure the ACME CA that issued the certificate is configured)"}`
`2024/06/03 17:32:56.091 INFO    tls.cache.maintenance   started background certificate maintenance      {"cache": "0xc000234400"}`
`2024/06/03 17:32:56.091 INFO    tls     storage cleaning happened too recently; skipping for now        {"storage": "FileStorage:/root/.local/share/caddy", "instance": "c06cb599-39cc-4e87-a53b-64648b791cf6", "try_again": "2024/06/04 17:32:56.091", "try_again_in": 86399.999999789}`
`2024/06/03 17:32:56.091 INFO    tls     finished cleaning storage units`

When I try to access https://vw.mydomain.xyz from my browser I get the following error in the Caddy log.

`2024/06/03 17:37:40.732 ERROR   http.log.error  dial tcp 127.0.0.1:8080: connect: connection refused    {"request": {"remote_ip": "10.19.88.101", "remote_port": "63821", "client_ip": "10.19.88.101", "proto": "HTTP/2.0", "method": "GET", "host": "vw.mydomain.xyz", "uri": "/", "headers": {"Sec-Ch-Ua-Platform": ["\"Windows\""], "Upgrade-Insecure-Requests": ["1"], "Accept": ["text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.7"], "Sec-Fetch-User": ["?1"], "Accept-Language": ["en-CA,en-GB;q=0.9,en-US;q=0.8,en;q=0.7"], "Priority": ["u=0, i"], "Sec-Ch-Ua": ["\"Google Chrome\";v=\"125\", \"Chromium\";v=\"125\", \"Not.A/Brand\";v=\"24\""], "Sec-Ch-Ua-Mobile": ["?0"], "User-Agent": ["Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/125.0.0.0 Safari/537.36"], "Sec-Fetch-Site": ["none"], "Sec-Fetch-Mode": ["navigate"], "Sec-Fetch-Dest": ["document"], "Accept-Encoding": ["gzip, deflate, br, zstd"]}, "tls": {"resumed": false, "version": 772, "cipher_suite": 4865, "proto": "h2", "server_name": "vw.mydomain.xyz"}}, "duration": 0.000341683, "status": 502, "err_id": "3a00n2p5b", "err_trace": "reverseproxy.statusError (reverseproxy.go:1269)"}`

**Comment by samuellhwarren** [comment]
You are pointing your domain to an http port, did you port forward that port from your router?

From: colin19889 ***@***.***>
Sent: Monday, June 3, 2024 10:48 AM
To: dani-garcia/vaultwarden ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [dani-garcia/vaultwarden] Private Instance with Let's Encrypt Certs (Proxmox LXC) - Issues with accessing via HTTPS (Discussion #4608)


I'm in the process of moving my Vaultwarden instance from my Synology NAS to my Proxmox Server. I have Vaultwarden installed on a Linux LXC (using tteck's script<https://helper-scripts.com/scripts?id=Vaultwarden>)and have migrated my existing database. I can access the web UI from the local IP but want to use my existing Cloudflare domain to access it via HTTPS (local only). When it was running on my Synology I used DuckDNS and Let's Encrypt but now want to use Cloudflare. I've followed this guide<https://github.com/dani-garcia/vaultwarden/wiki/Running-a-private-vaultwarden-instance-with-Let's-Encrypt-certs> with Caddy installed on the same LXC. I'm getting errors, and not being familiar with Caddy, I'm not entirely sure what is going on. My Caddyfile, caddy.env and output logs from Caddy are as follows. Is someone able to point me in the right direction? (Domain, cert_hash, unique_id etc.) have been removed.

My Caddyfile contains

{$DOMAIN}:443 { tls { dns cloudflare {$CLOUDFLARE_API_TOKEN} } reverse_proxy localhost:8080 }

My caddy.env file contains

DOMAIN=vw.mydomain.xyz CLOUDFLARE_API_TOKEN=my_api_token

The output logs from Caddy are as follows,

***@***.******@***.***:~/.local/share/caddy#> caddy run --envfile /root/.local/share/caddy/caddy.env 2024/06/03 17:32:56.085 INFO using adjacent Caddyfile 2024/06/03 17:32:56.086 INFO adapted config to JSON {"adapter": "caddyfile"} 2024/06/03 17:32:56.086 WARN Caddyfile input is not formatted; run 'caddy fmt --overwrite' to fix inconsistencies {"adapter": "caddyfile", "file": "Caddyfile", "line": 2} 2024/06/03 17:32:56.089 INFO admin admin endpoint started {"address": "localhost:2019", "enforce_origin": false, "origins": ["//localhost:2019", "//[::1]:2019", "//127.0.0.1:2019"]} 2024/06/03 17:32:56.089 INFO http.auto_https server is listening only on the HTTPS port but has no TLS connection policies; adding one to enable TLS {"server_name": "srv0", "https_port": 443} 2024/06/03 17:32:56.089 INFO http.auto_https enabling automatic HTTP->HTTPS redirects {"server_name": "srv0"} 2024/06/03 17:32:56.089 INFO http enabling HTTP/3 listener {"addr": ":443"} 2024/06/03 17:32:56.089 INFO failed to sufficiently increase receive buffer size (was: 208 kiB, wanted: 7168 kiB, got: 416 kiB). See https://github.com/quic-go/quic-go/wiki/UDP-Buffer-Sizes for details. 2024/06/03 17:32:56.089 INFO http.log server running {"name": "srv0", "protocols": ["h1", "h2", "h3"]} 2024/06/03 17:32:56.089 INFO http.log server running {"name": "remaining_auto_https_redirects", "protocols": ["h1", "h2", "h3"]} 2024/06/03 17:32:56.089 INFO http enabling automatic TLS certificate management {"domains": ["vw.mydomain.xyz"]} 2024/06/03 17:32:56.090 INFO autosaved config (load with --resume flag) {"file": "/root/.config/caddy/autosave.json"} 2024/06/03 17:32:56.090 INFO serving initial configuration 2024/06/03 17:32:56.090 WARN tls ARI window or selected renewal time changed {"identifiers": ["vw.mydomain.xyz"], "cert_hash": "HASH_HERE", "ari_unique_id": "UNIQUE_ID_HERE", "cert_expiry": "2024/08/29 18:22:51.000", "prev_start": "2024/07/29 18:42:51.000", "next_start": "0001/01/01 00:00:00.000", "prev_end": "2024/07/31 18:42:51.000", "next_end": "0001/01/01 00:00:00.000", "prev_selected_time": "2024/07/31 17:27:59.000", "next_selected_time": "0001/01/01 00:00:00.000", "explanation_url": ""} 2024/06/03 17:32:56.090 ERROR tls updating ARI upon managing {"error": "could not fully update ACME renewal info: either no ACME issuer configured for certificate, or all failed (make sure the ACME CA that issued the certificate is configured)"} 2024/06/03 17:32:56.091 INFO tls.cache.maintenance started background certificate maintenance {"cache": "0xc000234400"} 2024/06/03 17:32:56.091 INFO tls storage cleaning happened too recently; skipping for now {"storage": "FileStorage:/root/.local/share/caddy", "instance": "c06cb599-39cc-4e87-a53b-64648b791cf6", "try_again": "2024/06/04 17:32:56.091", "try_again_in": 86399.999999789} 2024/06/03 17:32:56.091 INFO tls finished cleaning storage units

When I try to access https://vw.mydomain.xyz from my browser I get the following error in the Caddy log.

2024/06/03 17:37:40.732 ERROR http.log.error dial tcp 127.0.0.1:8080: connect: connection refused {"request": {"remote_ip": "10.19.88.101", "remote_port": "63821", "client_ip": "10.19.88.101", "proto": "HTTP/2.0", "method": "GET", "host": "vw.mydomain.xyz", "uri": "/", "headers": {"Sec-Ch-Ua-Platform": ["\"Windows\""], "Upgrade-Insecure-Requests": ["1"], "Accept": ["text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.7"], "Sec-Fetch-User": ["?1"], "Accept-Language": ["en-CA,en-GB;q=0.9,en-US;q=0.8,en;q=0.7"], "Priority": ["u=0, i"], "Sec-Ch-Ua": ["\"Google Chrome\";v=\"125\", \"Chromium\";v=\"125\", \"Not.A/Brand\";v=\"24\""], "Sec-Ch-Ua-Mobile": ["?0"], "User-Agent": ["Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/125.0.0.0 Safari/537.36"], "Sec-Fetch-Site": ["none"], "Sec-Fetch-Mode": ["navigate"], "Sec-Fetch-Dest": ["document"], "Accept-Encoding": ["gzip, deflate, br, zstd"]}, "tls": {"resumed": false, "version": 772, "cipher_suite": 4865, "proto": "h2", "server_name": "vw.mydomain.xyz"}}, "duration": 0.000341683, "status": 502, "err_id": "3a00n2p5b", "err_trace": "reverseproxy.statusError (reverseproxy.go:1269)"}

**Comment by dfunkt** [comment]
I see that you use port 8080 in the caddy config, but the link with the install script specified port 8000. Did you by chance mistype it or have you actually changed it?
![Screenshot from 2024-06-03 21-55-25](https://github.com/dani-garcia/vaultwarden/assets/17910432/0ca037f7-e1a7-4a1e-bc98-bf66187ab49d)

**Reply by colin19889** [reply]
> I see that you use port 8080 in the caddy config, but the link with the install script specified port 8000. Did you by chance mistype it or have you actually changed it? ![Screenshot from 2024-06-03 21-55-25](https://private-user-images.githubusercontent.com/17910432/336194694-0ca037f7-e1a7-4a1e-bc98-bf66187ab49d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTc0NDI4ODEsIm5iZiI6MTcxNzQ0MjU4MSwicGF0aCI6Ii8xNzkxMDQzMi8zMzYxOTQ2OTQtMGNhMDM3ZjctZTFhNy00YTFlLWJjOTgtYmY2NjE4N2FiNDlkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjAzVDE5MjMwMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQxYmQyNTAxMWQwMjEyOTEwYzZkMjUwZTY2NThhMmU1NzFkNDMyNDAwM2IyMDFmZGEyZmU4YWYyZDE0MGJkY2EmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.MmbOndB52GsXe-Q6rj7hiHVV_FoPzfRil4cfVLtDzfU)

Well F....Thank you internet stranger. I must have looked at those files 15-times and never realized I fat fingered it. Seems to work as it should now.
